com.reprezen.swagedit/src/com/reprezen/swagedit/editor/hyperlinks/JsonReferenceHyperlinkDetector.java

```diff
@@ -37,42 +49,102 @@ protected boolean canDetect(String basePath) {
 
 	@Override
 	protected IHyperlink[] doDetect(SwaggerDocument doc, ITextViewer viewer, HyperlinkInfo info, String basePath) {
-		String label = null;
+		if (info.text.matches(LOCAL_REF_PATTERN.pattern())) {
+			return doDetectLocalLink(doc, viewer, info);
+		} else {
+			return doDetectExternalLink(info);
+		}
+	}
+
+	private IHyperlink[] doDetectExternalLink(HyperlinkInfo info) {
```

Comment:
It will fail at this point https://github.com/RepreZen/SwagEdit/blob/task/120/com.reprezen.swagedit/src/com/reprezen/swagedit/editor/hyperlinks/JsonReferenceHyperlinkDetector.java#L120 because % is an illegal character for a URI.